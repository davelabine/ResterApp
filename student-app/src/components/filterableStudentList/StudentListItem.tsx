import * as React from 'react';
import { ButtonGroup, Button, Glyphicon } from 'react-bootstrap';
import { StudentData } from '../../types';

export interface StudentListItemProps {
    student: StudentData;
    onUpdateStudentClick: (student: StudentData) => void;
    onDeleteStudent: (skey: string) => void;
}

export class StudentListItem extends React.Component<StudentListItemProps, object> {
    
    constructor(props: StudentListItemProps) {
        super(props);
        this.handleUpdateClick = this.handleUpdateClick.bind(this);
        this.handleDeleteClick = this.handleDeleteClick.bind(this);
    }

    public handleUpdateClick(): void {
        this.props.onUpdateStudentClick(this.props.student);
    }

    public handleDeleteClick(): void {
        this.props.onDeleteStudent(this.props.student.skey);
    } 

    public render() {
        let student = this.props.student;
        return (
            <tr>
                <td>{student.name}</td>
                <td>{student.id}</td>
                <td>
                    <ButtonGroup>
                        <Button id="edit" onClick={this.handleUpdateClick}><Glyphicon glyph="pencil"/></Button>
                        <Button id="delete" onClick={this.handleDeleteClick}><Glyphicon glyph="trash"/></Button>
                    </ButtonGroup>
                </td>
            </tr>
        );
    }
}