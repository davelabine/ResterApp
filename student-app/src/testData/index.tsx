import { StudentData } from '../types';

export const STUDENT_DATA_EMPTY: StudentData = new StudentData();

export const STUDENT_DATA_BILLY: StudentData =  {
                                                    'skey': '8a80810d5dd89a3f015dd89ac8280000',
                                                    'id': 'Billy Bob',
                                                    'name': '1234',
                                                    'photo': {
                                                        'bucketName': 'resterapp-dev',
                                                        'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
                                                    }
                                                };

export const LIST_STUDENT_DATA_EMPTY: StudentData[] = [];

export const LIST_STUDENT_DATA_TWO: StudentData[] =  [
    {
        'skey': '8a80810d5dd89a3f015dd89ac8280000',
        'id': 'Billy Bob',
        'name': '1234',
        'photo': {
            'bucketName': 'resterapp-dev',
            'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
        }
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a252800001',
        'id': '503074',
        'name': 'Albert Russell',
        'photo': {
            'bucketName': 'resterapp-dev',
            'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
        }
    },
];

export const LIST_STUDENT_DATA_BIG: StudentData[] =  [
    {
        'skey': '8a80810d5dd89a3f015dd89ac8280000',
        'id': '1234',
        'name': 'Billy Bob',
        'photo': {
            'bucketName': 'resterapp-dev',
            'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
        }
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a252800001',
        'id': '503074',
        'name': 'Albert Russell',
        'photo': {
            'bucketName': 'resterapp-dev',
            'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
        }
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a259b2000e',
        'id': '942226',
        'name': 'Alfredo Cummings',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a2677f0027',
        'id': '824388',
        'name': 'Allen Russell',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a271fb003a',
        'id': '266192',
        'name': 'Andre McKenna',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a27f0d0052',
        'id': '932083',
        'name': 'Angelica Obrien',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a281c80057',
        'id': '682307',
        'name': 'Anna Washington',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a27c52004d',
        'id': '734552',
        'name': 'Antonio Fowler',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a25fb80019',
        'id': '615643',
        'name': 'Ashley Mcdaniel',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a25897000c',
        'id': '753238',
        'name': 'Beatrice Russell',
        'photo': {}
    },
];
