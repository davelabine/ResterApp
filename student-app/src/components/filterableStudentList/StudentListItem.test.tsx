import * as React from 'react';
import { mount, shallow} from 'enzyme';
import { StudentListItem } from './StudentListItem';
import * as studentTestData from '../../testData';
import toJson from 'enzyme-to-json';

describe('StudentListItem', () => {

  it('renders wiuuthout crashing and matches the last snapshot', () => {
    const item = shallow(
                        <StudentListItem
                          key={studentTestData.STUDENT_DATA_BILLY.skey}
                          student={studentTestData.STUDENT_DATA_BILLY}
                          onDeleteStudent={jest.fn()}
                        />);
    expect(toJson(item)).toMatchSnapshot();
  });

  it('calls the delete callback', () => {
    const onDelete = jest.fn();
    const item = shallow(
                        <StudentListItem
                          key={studentTestData.STUDENT_DATA_BILLY.skey}
                          student={studentTestData.STUDENT_DATA_BILLY}
                          onDeleteStudent={onDelete}
                        />);
    const input = item.find({ id: 'delete'});
    expect(input.length).toEqual(1);
    input.simulate('click', 1);
    expect(onDelete).toHaveBeenCalled();
  });
}); 