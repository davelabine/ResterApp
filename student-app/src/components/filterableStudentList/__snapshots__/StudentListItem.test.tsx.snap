// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentListItem renders wiuuthout crashing and matches the last snapshot 1`] = `
<tr>
  <td>
    Billy Bob
  </td>
  <td>
    1234
  </td>
  <td>
    <ButtonGroup
      block={false}
      bsClass="btn-group"
      justified={false}
      vertical={false}
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        id="edit"
        onClick={[Function]}
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="pencil"
        />
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        id="delete"
        onClick={[Function]}
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="trash"
        />
      </Button>
    </ButtonGroup>
  </td>
</tr>
`;
